app/services/report_scanner/puppet_report_scanner.rb

```diff
@@ -1,6 +1,14 @@
 module Foreman
   class PuppetReportScanner
     class << self
+      def identify_origin(report_data)
+        'Puppet' if puppet_report?(report_data['logs'] || [])
+      end
+
+      def add_reporter_data(report, report_data)
+        scan(report, report_data['logs'] || [])
+      end
+
       def scan(report, logs)
         if (is_puppet = puppet_report?(logs))
           report.origin = 'Puppet'
```

Comment:
Does this method still need to set origin? I think it is now handled in report_importer.

Reply:
Yeah it does not, I was just trying to show the easies approach for plugins to do this switch, but I agree, let's rewrite it :)

Reply:
I've left just a comment for better understanding